contributing.md

```diff
@@ -1,90 +1,207 @@
 
-Start the playground:
+Before you start, please [search](https://github.com/search?q=repo%3Afingerprintjs%2FBotD) for your topic.
+There is a chance it has already been discussed.
 
-```bash
-yarn dev:playground
-```
+When you create an issue, please provide all the information needed to reproduce your situation, it will help us solve your issue faster.
+If you want to share a piece of code or the library output with us, please wrap it in a ` ``` ` block and make sure you include all the information.
 
-After a successful build, you can open http://localhost:3000 in your browser.
+### Creating a pull request
 
-**Note:** Playground uses a shared hardcoded botd API token.
+If you want to fix a bug, create a signal source, or make any other code contribution, please [create a pull request](https://docs.github.com/en/get-started/exploring-projects-on-github/contributing-to-a-project).
 
-### How to build
+After you clone the repository, check the [Working with code](#working-with-code) section to learn how to run, check, and build the code.
+
+In order for us to review and accept your code contributions, please follow these rules:
+- Your code quality should be at least as good as the code you modify.
+- Your code style (syntax, naming, coding patterns, etc) should follow the BotD style.
+- All the new code should be covered with automated tests.
+- All the checks described in the [Working with code](#working-with-code) section must pass successfully.
+  You may create a draft pull request in this repository to run the checks automatically by GitHub Actions,
+  but the tests won't run on BrowserStack until a BotD maintainer approves them.
+- If you want to add a bot detector, follow the [How to add a bot detector](#how-to-add-a-bot-detector) instructions carefully.
+- The changes should be backward compatible, ensuring BotD users continue to use the library without any modifications.
+- Don't add dependencies (such as Node packages) unless necessary.
+- Don't make changes unrelated to the stated purpose of your pull request. Please strive to introduce as few changes as possible.
+- Don't change BotD code style, its TypeScript configuration, or other subjective things.
+
+If you want to do something more complex than fixing a small bug, or if you're not sure if your changes meet the project requirements, please [start a discussion](https://github.com/fingerprintjs/BotD/discussions/new/choose).
+We encourage starting a discussion if you want to propose violating a rule from this guide.
+Doing so ensures we discuss all opinions, creating a good contribution experience for everyone.
+
+### Helping with existing issues
+
+If you want to help, but don't know where to start, take a look at the ["help wanted" issues](https://github.com/fingerprintjs/BotD/labels/help%20wanted).
```

Comment:
@xnerhu Please mark the appropriate [issues](https://github.com/fingerprintjs/BotD/issues) with a <kbd>help wanted</kbd> label